src/UXClient/Components/ScatterPlot/ScatterPlot.ts

```diff
@@ -0,0 +1,256 @@
+                
+                let xMeasure = this.measures[0], yMeasure = this.measures[1] ;
+                let rMeasure = this.measures[2] !== undefined ? this.measures[2] : null;
+                
+                //Init scales
+                this.yScale = d3.scaleLinear()
+                    .range([this.chartHeight, 0])
+                    .domain([this.extents[yMeasure][0] - yOffsetPercentage,this.extents[yMeasure][1] + yOffsetPercentage]);
+
+                this.xScale = d3.scaleLinear()
+                    .range([0, this.chartWidth])
+                    .domain([this.extents[xMeasure][0] - xOffsetPercentage,this.extents[xMeasure][1] + xOffsetPercentage]); 
+                    
+                if(rMeasure != null){
+                    this.rScale = d3.scaleLinear()
+                    .range([1, 10])
+                    .domain([this.extents[rMeasure][0],this.extents[rMeasure][1]]);
+                } else{
+                    this.rScale = () => 4;
+                }  
+
+                // Create color scale for each aggregate key
+                this.initColorScale();
+                
+                // Draw axis
+                this.drawAxis();
+
+                // Draw data
+                let scatter = this.g.selectAll(".tsi-dot")
+                    .data(this.cleanData(this.chartComponentData.allValues))
+
+                let self = this;
+                
+                scatter
+                    .enter()
+                    .append("circle")
+                    .attr("class", "tsi-dot")
+                    .on("mouseover", function(d) {	
+                        self.drawTooltip(d, [d3.event.pageX, d3.event.pageY])
+                    })					
+                    .on("mouseout", function(d) {		
+                        self.tooltip.hide();
+                    })
+                    .attr("r", (d) => this.rScale(d.measures[rMeasure]))
+                    .merge(scatter)
+                    .transition()
+                    .duration(self.TRANSDURATION)
+                    .attr("cx", (d) => this.xScale(d.measures[xMeasure]))
+                    .attr("cy", (d) => this.yScale(d.measures[yMeasure]))
+                    .attr("fill", (d) => this.colorMap[d.aggregateKey](d.splitBy))
+                    .attr("stroke-width", "1px")
+                    .attr("fill-opacity", .7)
+                    .attr("stroke-opacity", 1)
+                    .attr("stroke", (d) => this.colorMap[d.aggregateKey](d.splitBy))
+
+                scatter.exit().remove();
+                
+            }
+
+            // Add Window Resize Listener
+            window.addEventListener("resize", () => {
+                let self = this;
+                if (!this.chartOptions.suppressResizeListener) {
+                    this.draw();
+                }
+            });
+        }   
+        this.draw();                            
+    }
+
+    private initColorScale(){
```

Comment:
verify this is the same as what we do in line chart and how that is tied to legend

Reply:
createSplitByColors mayhaps?